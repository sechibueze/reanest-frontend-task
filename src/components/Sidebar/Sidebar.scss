@import "../../assets/theme/colors.scss";

.sidebar {
  display: none;
  position: fixed;
  height: 100%;
  width: 18rem;
  top: 0;
  left: 0;
  background-color: $white;
  @media screen and (min-width: $mobile-breakpoint) {
    & {
      display: block;
      position: relative;
      margin-right: 0.5rem;
    }
  }
}
.sidebar-header {
  background-color: $blue;
  position: relative;
  padding: 1.2rem 0.5rem;
  padding-left: 2rem;
}
.sidebar-close-btn {
  position: absolute;
  top: 0;
  right: 0;
  padding: 0.4rem;
  border-radius: 0.3rem;
  background-color: #333;
  color: #ddd;
  text-align: center;
  margin: auto;
  opacity: 0.7;
  cursor: pointer;
  @media screen and (min-width: $mobile-breakpoint) {
    & {
      display: none;
    }
  }
}

.sidebar-user-container {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-end;
  column-gap: 0.8rem;
  background-color: inherit;

  .sidebar-user-image {
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
  }
  .sidebar-user-data {
    background-color: transparent;
  }
}
.sidebar-username,
.sidebar-user-id {
  color: $white;
  font-size: 1rem;
  padding-bottom: 0.5rem;
}
.sidebar-user-id {
  padding-bottom: 0rem;
}
.sidebar-header-cta {
  display: block;
  width: 50%;
  margin: 1.5rem auto;
  margin-bottom: 0rem;
  padding: 0.5rem 1rem;
  border: 1px solid $white;
  border-radius: 1rem;
  color: $white;
  font-size: 0.8rem;
  text-align: center;
}

.sidebar-menu {
  width: 100%;
  margin-top: 2.5rem;
  background-color: $white;

  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
}
.icon-border {
  border: 1px solid #ccc;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  padding: 0.5rem;
  opacity: 0.4;
}
.sidebar-link {
  width: 100%;
  padding: 1rem 0rem;
  font-size: 1.2rem;
  padding-left: 1.2rem;
  .sidebar-menu-name {
    color: $light-gray;
    padding-left: 0.5rem;
    font-size: 0.7rem;
  }
  &:hover {
    background-color: $body-bg;
  }
}
