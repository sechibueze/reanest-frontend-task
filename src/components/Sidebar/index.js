import { useState, useEffect } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import axios from "axios";
import "./Sidebar.scss";
import { testData } from "../../_data";
// import userThumbnailPlaceholder from "../../assets/images/reanest-board-desktop.jpg";
export const links = [
  {
    name: "Dashboard",
    icon: "home",
    link: "/",
    prefix: "fas",
  },
  {
    name: "Countries",
    icon: "globe",
    link: "/countries",
    prefix: "fas",
  },
  {
    name: "States",
    icon: "xing-square",
    link: "/statess",
    prefix: "fab",
  },
  {
    name: "Schools",
    icon: "graduation-cap",
    link: "/schools",
    prefix: "fas",
  },
  {
    name: "Faculties",
    icon: "delicious",
    link: "/faculty",
    prefix: "fab",
  },
  {
    name: "Students",
    icon: "openid",
    link: "/students",
    prefix: "fab",
  },
];
const Sidebar = ({ user }) => {
  const [sidebarMenu, setSidebarMenu] = useState(links);

  useEffect(() => {
    const fetchMenuLinks = async () => {
      const uri = "http://601d848abe5f340017a19c29.mockapi.io/menu";
      // Whatever happens, overide api response with hardcoded data
      // some icons names from api are not in free version of fontawesome
      axios
        .get(uri)
        .then(({ data }) => {
          console.log("result ", data);
          setSidebarMenu(links);
        })
        .catch((err) => {
          setSidebarMenu(links);
          console.log("err ", { err });
        });
    };
    fetchMenuLinks();
  }, []);
  return (
    <div className="sidebar">
      <div className="sidebar-header">
        <label htmlFor="menu-checkbox" className="sidebar-close-btn">
          <FontAwesomeIcon icon="times" color="#ddd" />
        </label>
        <div className="sidebar-user-container">
          <img
            src={"https://i.ibb.co/swkR93p/bramdejager-600x600.png"}
            alt="user-thumbnail"
            className="sidebar-user-image"
          />
          <div className="sidebar-user-data">
            <p className="sidebar-username"> {user && user.name} </p>
            <p className="sidebar-user-id">
              {" "}
              {user && user.id && user.id.title} : {user.id.value}{" "}
            </p>
          </div>
        </div>

        <a href="https://" className="sidebar-header-cta">
          View Profile
        </a>
      </div>

      <ul className="sidebar-menu">
        {sidebarMenu &&
          Array.isArray(sidebarMenu) &&
          sidebarMenu.length > 0 &&
          sidebarMenu.map((menu) => {
            return (
              <li className="sidebar-link" key={menu.link}>
                <span className="icon-border">
                  <FontAwesomeIcon
                    icon={[menu.prefix, menu.icon]}
                    size={12}
                    fixedWidth
                  />
                </span>
                <a href={menu.link} className="sidebar-menu-name">
                  {" "}
                  {menu.name}{" "}
                </a>
              </li>
            );
          })}
      </ul>
    </div>
  );
};

Sidebar.defaultProps = {
  user: testData.user,
};
export default Sidebar;
