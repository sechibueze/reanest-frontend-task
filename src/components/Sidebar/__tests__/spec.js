import { shallow } from "enzyme";
import Sidebar, { links } from "../index";
describe("<Sidebar /> renders correctly", () => {
  const component = shallow(<Sidebar />);
  const sidebarWrapper = component.find(".sidebar");
  const sidebarLinks = component.find(".sidebar-link");
  it("<Sidebar /> matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("<Sidebar /> was rendered correctly", () => {
    expect(sidebarWrapper.length).toEqual(1);
  });
  it("Menu items rendered correct count", () => {
    expect(sidebarLinks.length).toEqual(links.length);
  });
});
