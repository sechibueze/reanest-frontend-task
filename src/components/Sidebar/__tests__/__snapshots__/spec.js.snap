// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly <Sidebar /> matches snapshot 1`] = `ShallowWrapper {}`;
